### PR TITLE
Plugins: Use handler middleware from the SDK

### PR DESCRIPTION
**Pull request by marefr**
**What is this feature?**

No feature. Refactoring/migrating plugin client middlewares to handler middlewares from the plugin SDK, ref https://github.com/grafana/grafana-plugin-sdk-go/pull/1019.

**Why do we need this feature?**

To be able to share some common functionality between grafana and the SDK, mainly related to observability of plugins.

**Who is this feature for?**

Developers and operators

**Which issue(s) does this PR fix?**:

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: pkg/plugins/manager/client/client.go [hunk 1]
```diff
@@ -101,8 +102,11 @@ func (s *Service) CallResource(ctx context.Context, req *backend.CallResourceReq
 				removeConnectionHeaders(res.Headers)
 				removeHopByHopHeaders(res.Headers)
 				removeNonAllowedHeaders(res.Headers)
+			} else {
+				res.Headers = map[string][]string{}
 			}
 
+			proxyutil.SetProxyResponseHeaders(res.Headers)
```
Comment (andresmgot) [inline pkg/plugins/manager/client/client.go:109]: this change seems unrelated?

Comment (marefr) [reply pkg/plugins/manager/client/client.go:109]: Yes, discussed historically https://github.com/grafana/grafana/pull/67145#discussion_r1175537374, and devised to remove the middleware to spare me some refactoring :) This being a middleware it can easily be missed to be registered and re-introduce security vulnerability. 

You're probably right that I should keep this out from these changes 👍

Comment (andresmgot) [reply pkg/plugins/manager/client/client.go:109]: it's okay if it's intentional / tested

**Review by andresmgot — COMMENTED** [review]
FWIW LGTM, we just need the SDK changes first

**Comment by marefr** [comment]
> FWIW LGTM, we just need the SDK changes first

You mean a tagged version of the SDK, right?

**Comment by andresmgot** [comment]
> > FWIW LGTM, we just need the SDK changes first
> 
> You mean a tagged version of the SDK, right?

yes

**Review by andresmgot — APPROVED** [review]
LGTM
